bin/shutter

```diff
@@ -5535,19 +5536,32 @@ sub STARTUP {
 		my $image = $clipboard->wait_for_image;
 		if (defined $image) {
 
+			#folder to save
+			my $folder = Glib::filename_to_unicode($saveDir_button->get_filename) ;
+			if ($save_no_active->get_active) {
+				$folder = Shutter::App::Directories::get_cache_dir();
+			}
+			
 			#determine current file type (name - description)
 			$combobox_type->get_active_text =~ /(.*) -/;
-			my $filetype_value = $1;
+			my $filetype_value = $1;			
 			unless ($filetype_value) {
 				$sd->dlg_error_message($d->get("No valid filetype specified"), $d->get("Failed"));
 				fct_control_main_window('show');
 				return FALSE;
 			}
+			
+			#generate unique random filename
+			my $ug = Data::UUID->new;
+			my $uuid = $ug->create();
+			my $short = $ug->to_string($uuid);
 
-			#create tempfile
-			my ($tmpfh, $tmpfilename) = tempfile(UNLINK => 1);
-			$tmpfilename .= "." . $filetype_value;
-
+			#relative to abs
+			my $tmpfilename = $folder ."/" . $short ."." . $filetype_value;
```

Comment:
I'd add somewhere in the filename that it's from clipboard, otherwise user will just end up with lots of files with strange names

Reply:
Good point! Done!

Reply:
We could use `srand()` instead of an UUID, the probability of two random numbers being the same is very low. Also, even with the UUID, it happen that two UUIDs match if they are generated during different Shutter sessions.

Reply:
I went ahead and substituted the UUID with a random number. What do you think?

Reply:
> Also, even with the UUID, it happen that two UUIDs match if they are generated during different Shutter sessions.

Not really. It depends on the algorithm used (aka "UUID version"), but either way the probability of clash is 0 or at least much lower than what you did with random.

Reply:
Would you suggest to revert to the UUID approach? Or keep it like this? Or maybe just increase the random number range to further reduce the clash probability?

Reply:
I think this is okay.